packages/circuit/quri_parts/circuit/utils/circuit_drawer.py

```diff
@@ -0,0 +1,386 @@
+        gate_aa = _generate_gate_aa(gate, idx)
+        layer, row_idx, gate_map = _place_check(gate, depth, gate_map)
+        circuit_pictures[layer] = _write_gate_string(
+            gate_aa, row_idx, circuit_pictures[layer]
+        )
+
+    # Join ``circuit_pictures`` into one numpy array of strings.
+    interm_layer = _interm_layer(qubit_count)
+    circuit_picture = interm_layer
+    for i, pict in enumerate(circuit_pictures):
+        circuit_picture = np.concatenate([circuit_picture, pict], axis=1)
+        if i != len(circuit_pictures) - 1:
+            circuit_picture = np.concatenate([circuit_picture, interm_layer], axis=1)
+
+    # Fill in the blanks.
+    circuit_picture = _connect_wire(circuit_picture)
+
+    # Divide and fold.
+    if circuit_picture.shape[1] > line_length:
+        n_divs = circuit_picture.shape[1] // line_length
+        output = circuit_picture[:, :line_length]
+        for i in range(1, n_divs):
+            output = np.concatenate(
+                [
+                    output,
+                    np.array(
+                        [
+                            [""] * line_length,
+                            [""] * line_length,
+                            ["="] * line_length,
+                            [""] * line_length,
+                        ]
+                    ),
+                ],
+                axis=0,
+            )
+            output = np.concatenate(
+                [
+                    output,
+                    circuit_picture[:, line_length * i : line_length * (i + 1)],  # noqa
+                ],
+                axis=0,
+            )
+    else:
+        output = circuit_picture
+
+    for line in output:
+        print("".join(line))
+
+
+def _generate_gate_aa(
+    gate: Union[ParametricQuantumGate, QuantumGate], gate_idx: int
+) -> Sequence[str]:
+    if gate_idx > 999:
+        warnings.warn(
+            f"Gate index larger than 999 is not supported. Index {gate_idx}"
+            " is modified to 999.",
+            Warning,
+        )
+    t_idxs = sorted(gate.target_indices)
+    c_idxs = sorted(gate.control_indices)
+    gate_string: list[str] = []
+
+    if gate.name == gate_names.SWAP:
+        gate_string = _create_swap_string(gate.target_indices, gate_idx)
+    else:
+        # Checks if control bit index < target bit index.
+        if len(c_idxs) != 0 and min(t_idxs) > min(c_idxs):
+            is_target_top = False
+            gate_string = _create_upper_control_part(gate_string, c_idxs, t_idxs)
+        else:
+            is_target_top = True
+
+        # Generate target part.
+        gate_size = max(t_idxs) - min(t_idxs) + 1
+        if is_target_top:
+            gate_head = "  ___  "
+        else:
+            gate_head = "  _|_  "
+        if gate.name in _GATE_STR_MAP:
+            g_str = _GATE_STR_MAP[gate.name].ljust(3)
+            gate_name_part = f" |{g_str}| "
+        else:
+            gate_name_part = " |UDF| "  # undefined gate
```

Comment:
Slightly shorter and easier to read?

```suggestion
         g_str = _GATE_STR_MAP.get(gate.name, "UDF").ljust(3)
         gate_name_part = f" |{g_str}| "
```